runtime/vm/createramclass.cpp

```diff
@@ -3655,16 +3669,17 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 						}
 					}
 
-					if (findModule) {
-						U_32 pkgNameLength = (U_32) packageNameLength(romClass);
-
-						omrthread_monitor_enter(javaVM->classLoaderModuleAndLocationMutex);
-						module = findModuleForPackage(vmThread, classLoader, J9UTF8_DATA(className), pkgNameLength);
-						omrthread_monitor_exit(javaVM->classLoaderModuleAndLocationMutex);
-					} else {
+					if (!findModule) {
 						module = javaVM->unamedModuleForSystemLoader;
 					}
 				}
+				if (findModule) {
+					U_32 pkgNameLength = (U_32)packageNameLength(romClass);
+					omrthread_monitor_t classLoaderModuleAndLocationMutex = javaVM->classLoaderModuleAndLocationMutex;
+					omrthread_monitor_enter(classLoaderModuleAndLocationMutex);
+					module = findModuleForPackage(vmThread, findModuleClassLoader, J9UTF8_DATA(className), pkgNameLength);
+					omrthread_monitor_exit(classLoaderModuleAndLocationMutex);
+				}
```

Comment:
An assertion could be added when `findModule` is false: 
`Assert_VM_true((module == javaVM->unamedModuleForSystemLoader) || (module == hostClass->module))`.

Reply:
Assertion has been added.